ui_testing/testcases/basic_tests/test05_testplans.py

```diff
@@ -51,8 +51,417 @@ def test001_test_plan_delete_testunit(self):
+
+        self.base_selenium.LOGGER.info('Searching for test plans with Completed status')
+        completed_testplans = self.test_plan_api.get_completed_testplans()
+
+        if completed_testplans is not None:
+            self.base_selenium.LOGGER.info('Getting the first testplan')
+            completed_testplan = completed_testplans[0]
+            old_completed_testplan_name = completed_testplan['testPlanName']
+            old_completed_testplan_version = completed_testplan['version']
+            self.base_selenium.LOGGER.info(
+                'Navigating to edit page of testplan: {} with version: {}'.format(old_completed_testplan_name,
+                                                                                  old_completed_testplan_version))
+            self.test_plan.get_test_plan_edit_page(name=old_completed_testplan_name)
+
+            # go to step 2 and add testunit
+            self.base_selenium.LOGGER.info('Going to step 2 to add testunit to this test plan')
+            self.test_plan.set_test_unit(test_unit='a')
+            self.test_plan.save_and_confirm_popup()
+
+            # go back to the active table
+            self.test_plan.get_test_plans_page()
+
+            # get the testplan to check its version
+            self.base_selenium.LOGGER.info('Getting the currently changed testplan to check its status and version')
+            inprogress_testplan_version, testplan_row_data_status = self.test_plan.get_testplan_version_and_status(
+                search_text=old_completed_testplan_name)
+
+            # self.assertEqual(old_completed_testplan_version + 1, int(inprogress_testplan_version))
+            self.assertGreater(int(inprogress_testplan_version), old_completed_testplan_version)
+            self.assertEqual(testplan_row_data_status, 'Completed')
+
+    def test004_archive_restore_test_plan_one_record(self):
+        '''
+        LIMS-3506 Case 1
+        Archive and restore one record
+        '''
+
+        self.base_selenium.LOGGER.info('Choosing a random testplan table row')
+        row = self.test_plan.get_random_table_row('test_plans:test_plans_table')
+        row_data = self.base_selenium.get_row_cells_dict_related_to_header(row=row)
+        testplan_number = row_data['Test Plan No.']
+        self.test_plan.sleep_small()
+
+        # archive and navigate to archived table
+        self.base_selenium.LOGGER.info('Testplan number: {} will be archived'.format(testplan_number))
+
+        self.base_selenium.LOGGER.info('Selecting the row')
+        self.test_plan.click_check_box(source=row)
+        self.test_plan.sleep_small()
+
+        self.test_plan.archive_selected_items()
+        self.test_plan.get_archived_items()
+
+        archived_row = self.test_plan.search(testplan_number)
+        self.test_plan.sleep_small()
+        self.base_selenium.LOGGER.info('Checking if testplan number: {} is archived correctly'.format(testplan_number))
+        self.assertIsNotNone(archived_row[0])
```

Comment:
assert the archived_row has the same number of the test plan

Reply:
In line 153 I'm searching by the testplan number, so I'm asserting that the archived_row variable isn't None, if it's None, then it wasn't archived correctly, else it's correct.

Reply:
not accurate, searching will return anything with this number even if there is another test case with a name equal to this number.

Reply:
I'll use filter instead.